bionic/decorators.py

```diff
@@ -103,7 +103,33 @@ def memoize(enabled):
     if not isinstance(enabled, bool):
         raise ValueError(f"Argument must be a boolean; got {enabled!r}")
 
-    return provider_wrapper(AttrUpdateProvider, 'should_memoize', enabled)
+    return provider_wrapper(AttrUpdateProvider, '_should_memoize', enabled)
+
+
+def volatile(enabled):
+    """
+    Indicates that the entity should always be recomputed between different
```

Comment:
There are two things we can talk about here:

1. What this decorator says about the decorated function: namely, that it is non-deterministic.
2. What behavior is caused by this decorator: namely, that this entity will be recomputed even if its inputs don't change.

I would start with 1, because it's more fundamental. That's also why I suggested the name `@volatile` (a fact about the function) instead of, say, `@always_recompute` (a desired behavior).

We should also discuss 2, but I'm not sure that "in the same flow" and "between different flows" are clear enough. It's not obvious whether "different flows" means "two different flow objects" or "flows from totally different Python files". I also think the key point for 2 is that the entity will be recomputed if its inputs don't change; the same/different flow thing is secondary (but still important and worth mentioning).

WDYT?

Reply:
I agree about non-deterministic and recompute regardless of input parts. 

I still think re-computation between flows is important enough to mention. I can see the confusion between flows and flow objects, I can reword it to say that the volatile entity is not recomputed within the same flow object. 

How does this sound?

Reply:
Oh, I totally agree that we should explain what happens with different flow objects. I would just structure the explanation so that the most foundational parts come first.